### PR TITLE
tikv: update docs about in-memory pessimistic locks

### PR DESCRIPTION
**Pull request by sticnarf**
Signed-off-by: Yilin Chen <sticnarf@gmail.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by en-jin19** [comment]
@sticnarf could you please invite a tech reviewer? thank you!

**Review by youjiali1995 — APPROVED** [review]

**Comment by sticnarf** [comment]
@youjiali1995 I add one more section in the pessimistic transaction page. Please take another look. Thanks!

**Review by youjiali1995 — APPROVED** [review]

**Comment by sticnarf** [comment]
Could you take a look? @en-jin19

**Comment by you06** [comment]
@en-jin19 PTAL

File: pessimistic-transaction.md [hunk 1]
```diff
@@ -177,3 +177,26 @@ pipelined = false
 ```sql
 set config tikv pessimistic-txn.pipelined='false';
 ```
+
+## 内存悲观锁
+
+TiKV 6.0 引入了内存悲观锁功能。开启内存悲观锁功能后，悲观锁通常只会被存储在 region leader 的内存中，而不会将锁持久化到磁盘，也不会通过 Raft 协议将锁同步到其他副本。当内存悲观锁占用的内存达到 region 或节点的阈值时，加悲观锁会回退为使用 [pipelined 加锁流程](#pipelined-加锁流程)。当 region 发生合并或 leader 迁移时，为避免悲观锁丢失，TiKV 会将内存悲观锁写入磁盘并同步到其他副本。
```
Comment (TomShawn) [inline pessimistic-transaction.md:183]:
```suggestion
TiKV 在 v6.0 中引入了内存悲观锁功能。开启内存悲观锁功能后，悲观锁通常只会被存储在 Region leader 的内存中，而不会将锁持久化到磁盘，也不会通过 Raft 协议将锁同步到其他副本。当内存悲观锁占用的内存达到 Region 或节点的阈值时，加悲观锁会回退为使用 [pipelined 加锁流程](#pipelined-加锁流程)。当 Region 发生合并或 leader 迁移时，为避免悲观锁丢失，TiKV 会将内存悲观锁写入磁盘并同步到其他副本。
```

Comment (TomShawn) [reply pessimistic-transaction.md:183]: 可否说一下使用内存悲观锁有什么好处吗？

File: pessimistic-transaction.md [hunk 2]
```diff
@@ -177,3 +177,26 @@ pipelined = false
...
+
+内存悲观锁实现了和 [pipelined 加锁](#pipelined-加锁流程)类似的表现，即集群无异常时不影响加锁表现，但当 TiKV 出现网络隔离或者节点宕机时，事务加的悲观锁可能丢失。
+
+如果业务逻辑依赖加锁或等锁机制，或者即使在集群异常情况下也要尽可能保证事务提交的成功率，应**关闭**内存悲观锁功能。
+
+该功能默认开启，可修改 TiKV 配置关闭：
```
Comment (TomShawn) [inline pessimistic-transaction.md:189]:
```suggestion
该功能默认开启。如要关闭，可修改 TiKV 配置：
```

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -1569,3 +1569,9 @@ Raft Engine 相关的配置项。
 
 + 开启流水线式加悲观锁流程。开启该功能后，TiKV 在检测数据满足加锁要求后，立刻通知 TiDB 执行后面的请求，并异步写入悲观锁，从而降低大部分延迟，显著提升悲观事务的性能。但有较低概率出现悲观锁异步写入失败的情况，可能会导致悲观事务提交失败。
 + 默认值：true
+
+### `in-memory` <span class="version-mark">从 v6.0 版本开始引入</span>
```
Comment (TomShawn) [inline tikv-configuration-file.md:1573]:
```suggestion
### `in-memory`（从 v6.0 版本开始引入）
```

File: pessimistic-transaction.md [hunk 4]
```diff
@@ -177,3 +177,28 @@ pipelined = false
 ```sql
 set config tikv pessimistic-txn.pipelined='false';
 ```
+
+## 内存悲观锁
+
+TiKV 在 v6.0 中引入了内存悲观锁功能。开启内存悲观锁功能后，悲观锁通常只会被存储在 Region leader 的内存中，而不会将锁持久化到磁盘，也不会通过 Raft 协议将锁同步到其他副本，因此可以大大降低悲观事务加锁的开销，提升悲观事务的吞吐并降低延迟。
```
Comment (TomShawn) [inline pessimistic-transaction.md:183]:
```suggestion
TiKV 在 v6.0.0 中引入了内存悲观锁功能。开启内存悲观锁功能后，悲观锁通常只会被存储在 Region leader 的内存中，而不会将锁持久化到磁盘，也不会通过 Raft 协议将锁同步到其他副本，因此可以大大降低悲观事务加锁的开销，提升悲观事务的吞吐并降低延迟。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Comment by TomShawn** [comment]
@ran-huang 新增术语”内存悲观锁/in-memory pessimistic lock“：

zh：开启内存悲观锁功能后，悲观锁通常只会被存储在 Region leader 的内存中，而不会将锁持久化到磁盘，也不会通过 Raft 协议将锁同步到其他副本，因此可以大大降低悲观事务加锁的开销，提升悲观事务的吞吐并降低延迟。

en： When this feature is enabled, pessimistic locks are usually stored in the memory of the Region leader only, and are not persisted to disk or replicated through Raft to other replicas. This feature can greatly reduce the overhead of acquiring pessimistic locks and improve the throughput of pessimistic transactions.

**Pull request merged**